core/src/jsMain/kotlin/Peripheral.kt

```diff
@@ -125,28 +125,28 @@ public class JsPeripheral internal constructor(
     private val gatt: BluetoothRemoteGATTServer
         get() = bluetoothDevice.gatt!! // fixme: !!
 
-    private val connectJob: SharedRepeatableTask = scope.sharedRepeatableTask {
+    private val connectJob: SharedRepeatableTask<Unit> = scope.sharedRepeatableTask {
         try {
             openConnection()
         } catch (cause: Throwable) {
             logger.error(cause) { message = "Failed to connect" }
-            disconnectJob.launch()
+            @Suppress("DeferredResultUnused") // Safe to ignore Deferred because the result is shared elsewhere
+            disconnectJob.getOrAsync()
             throw IOException("Connection attempt failed", cause)
         }
     }
 
-    private val disconnectJob: SharedRepeatableTask = scope.sharedRepeatableTask {
-        connectJob.join()
+    private val disconnectJob: SharedRepeatableTask<Unit> = scope.sharedRepeatableTask {
+        connectJob.cancelAndJoin()
```

Comment:
It sounds silly, but it took a long time to find the right combination of cancel/join of these jobs during both explicit and internally triggered disconnect events. Can we please keep this PR to modify the job `await` stuff only, and move any other changes to a separate PR for testing.

I should have added comments at the time to explain why it is like it is, sorry about that. This innocuous looking code is quite brittle.

Strongly suggest to keep `connectJob.join()` here and move to separate PR.